src/ephemeris.c

```diff
@@ -11,144 +12,147 @@
  */
 
 #include "linear_algebra.h"
+#include "constants.h"
 #include "ephemeris.h"
 #include <math.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <assert.h>
 
-#define NAV_OMEGAE_DOT 7.2921151467e-005
-#define NAV_GM 3.986005e14
-
-int calc_sat_pos(double pos[3], double vel[3],
-             double *clock_err, double *clock_rate_err,
-             const ephemeris_t *ephemeris,
-             gps_time_t tot)
+/** Calculate satellite position, velocity and clock offset from ephemeris.
+ *
+ * References:
+ *   -# IS-GPS-200D, Section 20.3.3.3.3.1 and Table 20-IV
+ *
+ * \param t GPS time at which to calculate the satellite parameters
```

Comment:
nit: If the function is called "calc_sat_state" we might want to say "...at which to calculate the satellite state" instead of "...at which to calculate the satellite parameters" for consistency.